include/swift/AST/Builtins.def

```diff
@@ -844,6 +844,11 @@ BUILTIN_MISC_OPERATION_WITH_SILGEN(BuildDefaultActorExecutorRef,
 BUILTIN_MISC_OPERATION_WITH_SILGEN(BuildMainActorExecutorRef,
                                    "buildMainActorExecutorRef", "n", Special)
 
+/// isOnStack : (Builtin.NativeObject) -> Builtin.Int1
```

Comment:
Out of curiosity, what would be a use case for this builtin?

Reply:
The use case for this builtin is this:

In certain cases the stdlib wants to change which code runs statically depending on whether or not the optimizer by the end of the pipeline puts a variable on the heap or the stack. The specific example is the routine malloc_size which is used in array.

Reply:
We can use this to fix stack promotion of arrays by avoiding calling malloc_size() when they're on the stack